module/zfs/mmp.c

```diff
@@ -398,10 +388,10 @@ mmp_write_uberblock(spa_t *spa)
 		zfs_dbgmsg("SCL_STATE acquisition took %llu ns\n",
 		    (u_longlong_t)lock_acquire_time);
 
-	error = mmp_random_leaf(spa->spa_root_vdev, &vd);
-
 	mutex_enter(&mmp->mmp_io_lock);
 
+	error = mmp_next_leaf(spa, &spa->mmp_last_leaf);
```

Comment:
`spa->mmp_last_leaf` can then be changed to a local variable here.

Reply:
That should work.  I'll try it.

Reply:
Can you also include a version of your nice comment, https://github.com/zfsonlinux/zfs/pull/7953#issue-217924306 , in the source describing why we rotate over the leaves in this way.

Reply:
OK, those changes are made.

Reply:
So, who pushes the "resolve conversation" button?  Me, because I made the changes you requested?  Or you, after you look for yourself?

Reply:
I'll do it!